sqlmesh/core/test/definition.py

```diff
@@ -281,25 +283,109 @@ def __init__(
             models=models,
         )
 
+    def _execute_model(self) -> pd.DataFrame:
+        """Executes the python model and returns a DataFrame."""
+        return t.cast(
+            pd.DataFrame,
+            next(self.model.render(context=self.context, **self.body.get("vars", {}))),
+        )
+
     def runTest(self) -> None:
         if "query" in self.body["outputs"]:
             expected_df = pd.DataFrame.from_records(self.body["outputs"]["query"])
-            actual_df = next(
-                self.model.render(
-                    context=self.context,
-                    **self.body.get("vars", {}),
-                )
-            )
-            actual_df = t.cast(pd.DataFrame, actual_df)
+            actual_df = self._execute_model()
             actual_df.reset_index(drop=True, inplace=True)
             self.assert_equal(expected_df, actual_df)
 
 
+def generate_test(
+    model: Model,
+    input_queries: t.Dict[str, str],
+    models: t.Dict[str, Model],
+    engine_adapter: EngineAdapter,
+    project_path: Path,
+    overwrite: bool = False,
+    variables: t.Optional[t.Dict[str, str]] = None,
+    path: t.Optional[str] = None,
+    name: t.Optional[str] = None,
+) -> None:
+    """Generate a unit test fixture for a given model.
+
+    Args:
+        model: The model to test.
+        input_queries: Mapping of model names to queries. Each model included in this mapping
+            will be populated in the test based on the results of the corresponding query.
+        models: The context's models.
+        engine_adapter: The target engine adapter.
+        project_path: The path pointing to the project's root directory.
+        overwrite: Whether to overwrite the existing test in case of a file path collision.
+            When set to False, an error will be raised if there is such a collision.
+        variables: Key-value pairs that will define variables needed by the model.
+        path: The file path corresponding to the fixture, relative to the test directory.
+            By default, the fixture will be created under the test directory and the file name
+            will be inferred from the test's name.
+        name: The name of the test. This is inferred from the model name by default.
+    """
+    test_name = name or f"test_{model.view_name}"
+    path = path or f"{test_name}.yaml"
+
+    extension = path.split(".")[-1].lower()
+    if extension not in ("yaml", "yml"):
+        path = f"{path}.yaml"
+
+    fixture_path = project_path / c.TESTS / path
+    if not overwrite and fixture_path.exists():
+        raise ConfigError(
+            f"Fixture '{fixture_path}' already exists, make sure to set --overwrite if it can be safely overwritten."
```

Comment:
Just curious: can't we just append in this case?

Reply:
I guess we can merge the dicts in the test (i.e. update `inputs` and `query`), though that should probably require an action from the user right? Like `--append` or `--update`.